### DISCUSSION TITLE
`cmd /inspect.rs`: Add `ContractArtifactField::Abi.ts`

### DISCUSSION THREAD

**Discussion by 0xbuidler** [foundry-rs/foundry#9102] [Ideas]
The idea of this additional option would transform the `json` abi to `typescript` abi by providing an extra option `abi.ts` to `ContractArtifactField` that would print the abi as Typescript instead of json.

Example:

`ContractAbi.json`:
```
[
   ...abi
]
```

`ContractAbi.ts`:
```
const ContractAbi = [
   ...abi
] as const;

export { ContractAbi };
```

I understand the reason why we wouldn't want this as this would include some specific code for a specific language but as of today most of the web3 frameworks are all or almost using [abitype](https://github.com/wevm/abitype) under the hood so why not providing an extra option `abi.ts` that would print the abi as Typescript instead of just json.

Any opinions or suggestions ? If this could be accepted I could open a PR this week.

**Comment by zerosnacks** [comment]
Hi @sshmaxime thanks for your suggestion

`forge build --inspect` operates directly on the artifact and simply returns the field. I don't think it is within the scope of Foundry to generate bindings to individual languages. I think this feature would be more suited for https://abitype.dev.

**Reply by 0xbuidler** [reply]
Hey @zerosnacks, yeah that was my thought at first but there is currently a Typescript limitation https://github.com/microsoft/TypeScript/issues/32063. That being said, this was quick and fast suggestion to get things done quickly but I do understand why it wouldn't be ideal. 

Thanks for the feedback !
